src/main/java/org/motechproject/newebodac/repository/custom/impl/FieldConfigRepositoryImpl.java

```diff
@@ -0,0 +1,52 @@
+package org.motechproject.newebodac.repository.custom.impl;
+
+import java.lang.reflect.Field;
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.motechproject.newebodac.domain.BaseEntity;
+import org.motechproject.newebodac.domain.enums.EntityType;
+import org.motechproject.newebodac.exception.NewEbodacException;
+import org.motechproject.newebodac.repository.custom.FieldConfigRepositoryCustom;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FieldConfigRepositoryImpl implements FieldConfigRepositoryCustom {
+
+  private final EntityManager em;
+  private final CriteriaBuilder cb;
+  private static final Logger LOGGER = LoggerFactory.getLogger(FieldConfigRepositoryImpl.class);
+  
+  public FieldConfigRepositoryImpl(EntityManager em) {
+    this.em = em;
+    this.cb = em.getCriteriaBuilder();
+  }
+
+  /** Checks if value should be unique, then if already exists. */
+
+  @Override
+
+
+  public void checkIfUnique(EntityType entityType, BaseEntity entity, UUID id,
```

Comment:
I think you should remove id from the parameters and change `Field field` to `Object fieldValue`